torch_xla/core/dynamo_bridge.py

```diff
@@ -387,6 +433,9 @@ def optimized_mod(*args):
     nonlocal xla_args_need_update
     nonlocal skip_checking_input_sharding_threashold
 
+    original_device: torch.device = _get_input_arg_device(input_args)
+    args = _maybe_move_tensors_to_device(input_args, xm.xla_device())
```

Comment:
maybe check if `original_device` is cuda? otherwise we don't need to call `_maybe_move_tensors_to_device ` which will loop through all input tensors.

Reply:
See other comment, I don't think this should be limited to CUDA only unless you see a strong reason to?

Also, see @golechwierowicz comment thread where we measured the overhead which seemed negligible.